Add localregistry input option

lukasgrunwald opened the pull request:
Implements and resolves #37 .
Option to add local registries to build process by specifying their url (https). Authentication to the repos via specified GitHub token (default: User Github Token)

Snippet example for using the new flags, that adds two private Registries `CondMatRegistry.git` and `HolyLabRegistry.git`
```yaml
...
      - uses: lukasgrunwald/julia-buildpkg@localregistry
        with:
          localregistry: >- # Collect URL's into space seperated string
            https://github.com/lukasgrunwald/CondMatRegistry.git
            https://github.com/HolyLab/HolyLabRegistry.git
          token: ${{ secrets.GITHUB_TOKEN }} # Automatically generated token
...
```

SaschaMann commented:
Thanks!

Could you add the example to the README please?

I'm not sure about the input interface of `>-`. I've rarely seen that being used in workflows and I'm not sure most people are aware of its existance. If they copy and paste it, that's fine, but perhaps there's a more common way using `|`? That's how e.g. actions/cache handles similar list-like inputs.

lukasgrunwald commented:
Hey @SaschaMann,
I added an example to  the ReadMe and updated the interface to use "|". Additionally I dropped the `token` input option, since this is probably better set by the user in -actions/checkout@v4 anyway. Instead I added the `git_cli` input, which sets JULIA_PKG_USE_CLI_GIT = git_cli, giving the user the option to control the git setup used by Pkg.

SaschaMann commented on README.md:
```diff
@@ -43,3 +43,29 @@ is already set. If you want another registry flavor (i.e. `conservative`) this
+          ssh-private-key: |
+            ${{ secrets.PRIVATE_DEPLOY_KEY }}
+            ${{ secrets.PRIVATE_DEPLOY_KEY2 }}
+      - uses: julia-actions/julia-buildpkg@main # Update @main once new taged version available
```
```suggestion
      - uses: julia-actions/julia-buildpkg@v1
```

Let's tag a release immediately after merging, so we won't have to update this afterwards

SaschaMann commented on action.yml:
```diff
@@ -14,11 +14,13 @@ inputs:
     default: 'no'
   localregistry:
-    description: 'Add local registries hosted on GitHub. Speciefied by providing the url (https) to the repositories as a single-space seperated list'
+    description: 'Add local registries hosted on GitHub. Speciefied by providing the url (https/ssh) to the repositories as a newline (\n) seperated list.
```
```suggestion
    description: 'Add local registries hosted on GitHub. Specified by providing the url (https/ssh) to the repositories as a newline (\n) seperated list.
```

SaschaMann commented on action.yml:
```diff
@@ -28,13 +30,18 @@ runs:
   - run: |
       import Pkg
+      
+      # Determine if Pkg uses git-cli executable instead of LibGit2
```
Might be good to have a warning or at least info message if the input is set but the version is <= 1.6. Perhaps a warning is too much noise but at least an info that the input isn't being used is probably good?

lukasgrunwald replied:
Good point! Added it as a yellow print statement, so that it should catch the eye more easily :+1:

SaschaMann commented:
Thank you! Some minor notes but then it's ready to be merged :)

SaschaMann commented:
Thank you!

DilumAluthge commented:
@SaschaMann It looks like this will override a value of `JULIA_PKG_USE_CLI_GIT` set earlier in the workflow file by the user. Is this intentional?

IMO, if the user has already set the `JULIA_PKG_USE_CLI_GIT` environment variable, and they don't provide the `git_cli` input to this action, then we should respect the environment variable set by the user.

DilumAluthge commented:
> IMO, if the user has already set the `JULIA_PKG_USE_CLI_GIT` environment variable, and they don't provide the `git_cli` input to this action, then we should respect the environment variable set by the user.

#39
